### ISSUE TITLE
Idea: Rule-based actions on tabs

### ISSUE THREAD

**Issue by GitHubGeek** [deanoemcke/thegreatsuspender#118]
Hello,

This could be a very cool feature for power users, for managing a large number of new tabs efficiently and automatically.

_TL;DR: Allow users to automate tabs management by defining rules similar to "email inbox rules"_

User should be able to define filtering rules in The Great Suspender. User can then associate one or more actions to each of the "rules set". The rules engine in The Great Suspender runs the rules sets and perform the actions on behave of the user.

Possible rules for filtering browser tabs:
1. Tab has been suspended for over X hours / minutes in total
2. Tab has not been un-suspended in the last X hours / minutes
3. Site domain name
4. Page title contains keyword _Foo_
5. Page URL contains keyword _Foo_
6. Tab is pinned
7. There are more than X number of tabs opened in a window

User should be able to combine the above rules in any manner (AND/OR/NOT) in a _rules set_.

Possible actions:
1. Suspend this tab :smile: 
2. Never suspend this tab
3. Pin tab
4. Unpin tab
5. Close tab
6. Pop up notification with custom text
7. Screen capture
8. Add to bookmark folder _Foo_
9. Send page URL to email
10. Save page to Google Drive / Google Keep / Evernote / Dropbox
11. Send page to mobile device (via Chrome to Phone / PushBullet)
12. Download tab content (HTML) as a file
13. IFTTT
14. Clear all cookies of the site loaded in tab
15. Clear all cookies & cache of the site loaded in tab

Actions should be stackable, just like in Gmail / Outlook.

Thanks for reading!
